Libraries/xcassets/Headers/xcassets/Asset/Asset.h

```diff
@@ -29,6 +29,7 @@ class Asset {
 private:
     FullyQualifiedName         _name;
     std::string                _path;
+    libutil::Filesystem const *_filesystem;
```

Comment:
Hm, where in the catalog object? If it's synchronous with creation, you should be able to just pass it down as a parameter. If it's later on (e.g. a method) then it still has the same issues about requiring a specific lifetime for the filesystem.